### PR TITLE
Update/buttonfab

### PR DESCRIPTION
**Pull request by macandcheese**
@asangma this is a draft with a demo on the index example page of folks positioning on their own with props added to the base `<calcite-button>` component - `isFloating` and `isRound`.

Pros of this approach is that it's non-prescriptive with its positioning - folks can position their own wrapper as fixed, sticky, etc., based on their own needs, all existing button appearance options are available, and that it lets folks use `isRound` inline and independently of `isFloating`

Cons of this approach is that we run the risk of folks using `isFloating` without wrapping in their own positioned container, which would add "level 2 styling" on an inline element (note - we are updating the box-shadow levels slightly to make these floating items that are clickable a little more noticeable cc @bstifle @julio8a ), or using something weird like an "inline" button. We'd have to doc that.

### CODE REVIEW

File: src/components/calcite-button/calcite-button.tsx [hunk 1]
```diff
@@ -56,6 +56,12 @@ export class CalciteButton {
   /** optionally add a calcite-loader component to the button, disabling interaction.  */
   @Prop({ reflect: true }) loading?: boolean = false;
 
+  /** optionally add a rounded style to the button  */
+  @Prop({ reflect: true }) isRound?: boolean = false;
```
Comment (driskull) [inline src/components/calcite-button/calcite-button.tsx]: Can we just say `round/rounded` and `float/floated` instead of the `is`?

**Comment by asangma** [comment]
This seems to work.
I hear you on devs potentially misusing the floating style.  But I think we can take that risk.

Also along the lines of property naming, we had some discussion on a separate component that "float" may get confused with the ol' CSS `float` property.

I don't have a good alternative except maybe `lifted`. ¯\_(ツ)_/¯

lifted

**Comment by asangma** [comment]
I think there are some alignment issues with the icon.
image

Are we not able to use flex to center stuffs? :)

**Comment by asangma** [comment]
We have an [autoText prop](https://esri.github.io/calcite-app-components/demos/calcite-action.html) in `calcite-action`. Would that be a possibility?

**Comment by macandcheese** [comment]
Yeah agreed, all buttons need to have icon positioning improved, for now an issue is that we are reliant on a single icon path (and one that is passed in) that is used for all scale of button. 

When we add a `calcite-icon` component, we can accept a string, match it to an icon, and then better use one of the three sizes based on the button scale.

Feel free to take a stab at the "revealTextOnHover" - I've never reliably been able to get a smooth animation there. For a FAB maybe tooltips could solve for that?

Screen Shot 2019-12-03 at 10 45 08 AM

**Comment by macandcheese** [comment]
@asangma updated alignment, should be better until we can swap in scale-appropriate icons!

**Comment by macandcheese** [comment]
> Can we just say `round/rounded` and `float/floated` instead of the `is`?

> I don't have a good alternative except maybe `lifted`. ¯_(ツ)_/¯

`floated` and `rounded` ? `floating` and `round` ? `pill` and `hill` ???

I can update those and add this in, and include an example of the "custom fab wrapper" in the Storybook.

**Comment by macandcheese** [comment]
Updated nomenclature, waiting for an update to `calcite-base` and `calcite-colors` to do some styling updates, then I will move this out of draft. Any other nomenclature issues / feedback?

**Pull request merged**